### PR TITLE
Create information box in courts/index page

### PR DESCRIPTION
**Pull request by elijahkim**
- On Marker click, information box displays name and address
- On map click, information box hides
  It looks like this when its clicked on a marker 
  screen shot 2015-01-19 at 3 45 26 pm

and like this when you click on the map
screen shot 2015-01-19 at 3 46 22 pm

### CODE REVIEW

File: app/assets/javascripts/map.js [hunk 1]
```diff
@@ -4,16 +4,26 @@ $(function() {
...
+    map = L.mapbox.map('map', window.MAPBOX_USER_NAME, { zoomControl: false}).
       setView([latitude, longitude], 14);
   };
 
   var createMarker = function(court) {
     var marker = L.marker([ court.latitude, court.longitude], {
       icon: L.mapbox.marker.icon({
-        'marker-color': '#9c89cc'
+        'marker-color': '#9c89cc',
+        'clickable': true
```
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: what's this do?

File: app/assets/javascripts/map.js [hunk 2]
```diff
@@ -4,16 +4,26 @@ $(function() {
...
       })
-    }).addTo(map);
+    })
+    .addTo(map);
+    marker.on("click", function(){
```
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: O.o why aren't you using the normal Mapbox way of doing this?

File: app/assets/stylesheets/pages/_court.scss [hunk 3]
```diff
@@ -39,3 +39,20 @@
...
+  display: block;
+  background-color: rgba(0,0,0,0.5);
+  color: white;
+  div {
+    padding-left: .3em;
+  }
+  a {
+    padding-left: .3em;
+    color: $yellow;
+    &:hover{color: darken($yellow, 30%);}
```
Comment (HayleyCAnderson) [inline app/assets/stylesheets/pages/_court.scss]: whaaaat I didn't know you could do that

File: app/assets/javascripts/map.js (hunk 1 above)
Comment (elijahkim) [inline app/assets/javascripts/map.js]: I'm not sure if I need it or not. I'll see what happens.

File: app/assets/javascripts/map.js (hunk 2 above)
Comment (elijahkim) [inline app/assets/javascripts/map.js]: I'm not sure if I could do what I want otherwise. Take a look at the description

File: app/assets/stylesheets/pages/_court.scss (hunk 3 above)
Comment (elijahkim) [inline app/assets/stylesheets/pages/_court.scss]: Dustin helped me out on this

File: app/assets/javascripts/map.js (hunk 2 above)
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: wait, where are you defining court??

File: app/assets/javascripts/map.js [hunk 4]
```diff
@@ -4,16 +4,25 @@ $(function() {
...
-    map = L.mapbox.map('map', window.MAPBOX_USER_NAME).
+    map = L.mapbox.map('map', window.MAPBOX_USER_NAME, { zoomControl: false}).
       setView([latitude, longitude], 14);
   };
 
   var createMarker = function(court) {
     var marker = L.marker([ court.latitude, court.longitude], {
       icon: L.mapbox.marker.icon({
-        'marker-color': '#9c89cc'
+        'marker-color': '#9c89cc',
```
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: `"description" => "<p>content of your popup</p>"`

File: app/assets/stylesheets/pages/_court.scss (hunk 3 above)
Comment (HayleyCAnderson) [inline app/assets/stylesheets/pages/_court.scss]: awesome

File: app/assets/javascripts/map.js (hunk 2 above)
Comment (elijahkim) [inline app/assets/javascripts/map.js]: its getting passed from `createMarker`

File: app/assets/javascripts/map.js (hunk 4 above)
Comment (elijahkim) [inline app/assets/javascripts/map.js]: it's not a popup though

Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: ohhhh. Is it like, an informational box that pops up over the whole map?

Comment (elijahkim) [inline app/assets/javascripts/map.js]: yeah! I added photos on the description

Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: huh. cool. let me know when it's on heroku, I'd like to see it

File: app/assets/javascripts/map.js (hunk 2 above)
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: still confused, but as long as it works...

File: app/assets/javascripts/map.js (hunk 4 above)
Comment (elijahkim) [inline app/assets/javascripts/map.js]: Are you coming to project night? :)

File: app/assets/stylesheets/pages/_court.scss (hunk 3 above)
Comment (AbbyBiying) [inline app/assets/stylesheets/pages/_court.scss]: very cool

File: app/assets/javascripts/map.js (hunk 4 above)
Comment (HayleyCAnderson) [inline app/assets/javascripts/map.js]: yeah, I guess so

File: app/assets/javascripts/map.js [hunk 5]
```diff
@@ -4,16 +4,26 @@ $(function() {
 
   var createMap = function(latitude, longitude){
     L.mapbox.accessToken = window.MAPBOX_ACCESS_TOKEN;
-    map = L.mapbox.map('map', window.MAPBOX_USER_NAME).
+    map = L.mapbox.map('map', window.MAPBOX_USER_NAME, { zoomControl: false }).
```
Comment (dgalarza) [inline app/assets/javascripts/map.js]: Didn't catch this earlier, but should be using double quotes here. Also, probably should enable hound for javascript.

File: app/assets/javascripts/map.js [hunk 6]
```diff
@@ -4,16 +4,26 @@ $(function() {
...
       })
-    }).addTo(map);
+    })
+    .addTo(map);
+    marker.on("click", function(){
+      $(".information-box").html(
+          "<div>"+court.name+"</div>" +
+          "<div>"+court.address+"</div>" +
+          "<div>Average Elo: "+court.average_elo+"</div>"+
+          "<a href=/courts/"+court.id+">Go to Court</a>"
```
Comment (dgalarza) [inline app/assets/javascripts/map.js]: Not necessarily needed now, but could be nice to use something like [handlebars](http://handlebarsjs.com/) for the HTML templating.

File: app/assets/javascripts/map.js [hunk 7]
```diff
@@ -34,6 +44,9 @@ $(function() {
       url: "/api/courts",
     });
     courtsRequest.done(plotCourts);
+    map.on("click", function(){
```
Comment (dgalarza) [inline app/assets/javascripts/map.js]: Instead of using an anonymous function like this what about extracting another function called `hideInformationBox` to convey your intent.

File: app/assets/stylesheets/base/_variables.scss [hunk 8]
```diff
@@ -32,6 +32,7 @@ $light-yellow: #FFF6BF;
 $light-green: #E6EFC2;
 $orange: #FF6600;
 $grey: #E8E8E8;
+$yellow: #FBF764;
```
Comment (dgalarza) [inline app/assets/stylesheets/base/_variables.scss]: Consider alphabetizing this list.

File: app/assets/stylesheets/pages/_court.scss [hunk 9]
```diff
@@ -39,3 +39,20 @@
...
 }
+
+.information-box {
+  z-index: 1;
+  margin-left: 1em;
+  width: 20em;
+  display: block;
+  background-color: rgba(0,0,0,0.5);
+  color: white;
+  div {
```
Comment (dgalarza) [inline app/assets/stylesheets/pages/_court.scss]: > Use a blank line above a selector that has styles
https://github.com/thoughtbot/guides/tree/master/style/sass#formatting

File: app/assets/stylesheets/pages/_court.scss [hunk 10]
```diff
@@ -39,3 +39,20 @@
...
+    padding-left: .3em;
+  }
+  a {
```
Comment (dgalarza) [inline app/assets/stylesheets/pages/_court.scss]: > Use a blank line above a selector that has styles
https://github.com/thoughtbot/guides/tree/master/style/sass#formatting

**Comment by HayleyCAnderson** [comment]
@dgalarza ! Can you make suggestions on my terrible javascript??

File: app/assets/stylesheets/pages/_court.scss (hunk 3 above)
Comment (dgalarza) [inline app/assets/stylesheets/pages/_court.scss]: Avoid using a single line for a style like this

**Comment by AbbyBiying** [comment]
@dgalarza me too! I did not even know hound is not watching my JS! I need to fix my JS!

**Comment by dgalarza** [comment]
Hound supports both scss and javascript / coffeescript. Just need to enable it. See https://houndci.com/configuration for more info.

As far as looking at PRs. Send them my way and if I have some time I'll take a look :smile:

**Comment by HayleyCAnderson** [comment]
I've used Code Climate to fix my syntax problems that I didn't realize I had... but I'm more concerned about how it's actually being done. I don't have new JS PRs out but I didn't really do the best job on it when I was creating it and now I'm going back to try to improve performance throughout my app, but I don't have enough experience w/ JS to know how to make it much better...

**Comment by HayleyCAnderson** [comment]
TL;DR, could you just look at my JS files?
